retriever/lib/engine.py

```diff
@@ -521,34 +550,44 @@ def download_files_from_archive(
             if not os.path.exists(archive_dir):
                 os.makedirs(archive_dir)
 
-        if not file_names:
-            self.download_file(url, archive_name)
-            if archive_type in ('tar', 'tar.gz'):
-                file_names = self.extract_tar(archive_full_path, archive_dir,
-                                              archive_type)
-            elif archive_type == 'zip':
-                file_names = self.extract_zip(archive_full_path, archive_dir)
-            elif archive_type == 'gz':
-                file_names = self.extract_gz(archive_full_path, archive_dir)
-            return file_names
+        if url.split(":")[0] ==  "kaggle":
```

Comment:
The engine has an attribute of script. 
From which we get all the values like url and paths used in the downloading function.
We can check if that attribute of script has `kaggle` and if `kaggle is true`